src/test/java/com/box/sdk/BoxSignRequestTest.java

```diff
@@ -0,0 +1,256 @@
+        BoxSignRequestFile file = new BoxSignRequestFile(fileId);
+        files.add(file);
+
+        String signerEmail = "user@example.com";
+        List<BoxSignRequestSigner> signers = new ArrayList<BoxSignRequestSigner>();
+        BoxSignRequestSigner newSigner = new BoxSignRequestSigner(signerEmail)
+                .setInPerson(false);
+        signers.add(newSigner);
+
+        BoxFolder testFolder = new BoxFolder(api, "141841942473");
+        BoxFolder.Info parentFolderInfo = testFolder.createFolder("Test " + NumericDate.now().getValue());
+        String parentFolderId = parentFolderInfo.getID();
+
+        // Do Create
+        BoxSignRequestCreateParams createParams = new BoxSignRequestCreateParams()
+                .setIsDocumentPreparationNeeded(true);
+        BoxSignRequest.Info signRequestInfoCreate = BoxSignRequest.createSignRequest(api, files,
+                signers, parentFolderId, createParams);
+
+        String signRequestIdCreate = signRequestInfoCreate.getID();
+        BoxFile.Info fileInfoCreate = signRequestInfoCreate.getSourceFiles().get(0);
+
+        // Todo: get signer by role type. Using index=1 is fragile, as order may not be guaranteed.
+        //signer at index 0 has role=final_copy_reader
+        //signer at index 1 has role=signer
+        BoxSignRequestSigner signerCreate = signRequestInfoCreate.getSigners().get(1);
+
+        // Test Create
+        Assert.assertNotNull(signRequestInfoCreate.getPrepareUrl());
+        Assert.assertEquals(fileId, fileInfoCreate.getID());
+        Assert.assertEquals(signerEmail, signerCreate.getEmail());
+        Assert.assertNotNull(signRequestInfoCreate.getID());
+
+        // Do Get by ID
+        BoxSignRequest signRequestGetByID = new BoxSignRequest(api, signRequestIdCreate);
+        BoxSignRequest.Info signRequestInfoGetByID = signRequestGetByID.getInfo();
+        BoxFile.Info fileInfo = signRequestInfoGetByID.getSourceFiles().get(0);
+
+        // Todo: get signer by role type. Using index=1 is fragile, as order may not be guaranteed.
+        //signer at index 0 has role=final_copy_reader
+        //signer at index 1 has role=signer
+        BoxSignRequestSigner signer = signRequestInfoGetByID.getSigners().get(1);
+
+        // Test Get by ID
+        Assert.assertEquals(fileId, fileInfo.getID());
+        Assert.assertEquals(signerEmail, signer.getEmail());
+        Assert.assertEquals(signRequestIdCreate, signRequestInfoGetByID.getID());
+
+        // Do Get All
+        Iterable<BoxSignRequest.Info> signRequestsGetAll = BoxSignRequest.getAll(api);
+
+        // Test Get All
+        Assert.assertTrue(signRequestsGetAll.iterator().hasNext());
+
+        // Do Cancel
+        // Cancel will fail if it's too soon after creation
+        Thread.sleep(3000);
+        BoxSignRequest.Info signRequestInfoCancel = signRequestGetByID.cancel();
+        BoxSignRequest signRequestGetByIDAfterCancel = new BoxSignRequest(api, signRequestIdCreate);
```

Comment:
`signRequestGetByIDAfterCancel ` is not used anywhere. I guess the idea was to use it in line 236. We can do this or remove this line altogether.